### PR TITLE
Also track coverage information for C code

### PR DESCRIPTION
**Pull request by psychon**
Fixes: https://github.com/awesomeWM/awesome/issues/1364

### CODE REVIEW

**Comment by blueyed** [comment]
Thanks!
I think we can just add the `gcov` call to `do_codecov`?!
The first change will upload it for the existing Codecov flags, and the second change as `*_c`.

**Comment by psychon** [comment]
I'm just not sure if your change means that coverage is "accumulated" or if $SOMETHING deletes/overwrites the coverage information files when needed...

Also, your changes broke travis: /home/travis/.travis/job_stages: eval: line 71: syntax error near unexpected token `else'

File: .travis.yml [hunk 1]
```diff
@@ -195,9 +188,7 @@ script:
         (make check-unit-coverage \
          && do_codecov unittests \
          && tests/run.sh \
-         && do_codecov functionaltests \
-         && gcov $(find -name '*.o') \
-         && do_codecov_gcov c_code)
+         && do_codecov functionaltests
```
Comment (psychon) [inline .travis.yml]: Missing `)`, I think.

**Comment by blueyed** [comment]
Should be finally fixed.
I have kept the two commits, so we should be able to compare the outcome(s).
There should be `*_c`, so that the C coverage is also grouped/flagged like the Lua one.

**Comment by psychon** [comment]
(Codecov has some weird problem with the C files; I already reported that two days ago).

Fine by me, but I'm not sure how useful `samples_c` and `unittests_c` is since both do not run the awesome binary (and thus are much faster than functional tests...) (and for some weird reason, `unittests_c` and `samples_c` still get 0.74% coverage?!?).

**Comment by blueyed** [comment]
Let's use your initial approach.  I'll cherry-pick it to master.

**Comment by psychon** [comment]
Thanks for cherry-picking and thanks for looking into this.

**Pull request closed without merging**